server/lib/http_provider.py

```diff
@@ -0,0 +1,79 @@
+
+from .import_providers import ImportProvider
+from .entity import Entity
+from .data_map import DataMap
+from .file_map import FileMap
+
+
+class HTTPImportProvider(ImportProvider):
+    def __init__(self):
+        super().__init__('HTTP')
+
+    def matches(self, entity: Entity) -> bool:
+        value = entity.getValue()
+        return value.startswith('http://') or value.startswith('https://')
+
+    def lookup(self, entity: Entity) -> DataMap:
+        pid = entity.getValue()
+        url = urlparse(pid)
+        if url.scheme not in ('http', 'https'):
+            # This should be redundant. This should only be called if matches()
+            # returns True, which, various errors aside, signifies a commitment
+            # to the entity being legitimate from the perspective of this provider
+            raise Exception('Unknown scheme %s' % url.scheme)
+        headers = requests.head(pid).headers
+
+        valid_target = headers.get('Content-Type') is not None
+        valid_target = valid_target and ('Content-Length' in headers or
+                                         'Content-Range' in headers)
+        if not valid_target:
+            raise Exception('Failed to get size for %s' % pid)
+
+        if 'Content-Disposition' in headers:
+            fname = re.search('^.*filename=([\w.]+).*$',
+                              headers['Content-Disposition'])
+            if fname:
+                fname = fname.groups()[0]
+        else:
+            fname = os.path.basename(url.path.rstrip('/'))
+
+        size = headers.get('Content-Length') or \
+            headers.get('Content-Range').split('/')[-1]
+
+        return DataMap(pid, int(size), name=fname, repository=self.getName())
+
+    def listFiles(self, entity: Entity) -> FileMap:
+        dm = self.lookup(entity)
+        if dm is None:
+            return None
+        else:
+            fm = FileMap(dm.getName())
+            fm.addFile(entity.getValue(), dm.getSize())
+            return fm
+
+    def register(self, parent: object, parentType: str, progress, user, dataMap: DataMap,
+                 base_url: str=None):
+        url = dataMap.getDataId()
+        progress.update(increment=1, message='Processing file {}.'.format(url))
+        headers = requests.head(url).headers
+        size = headers.get('Content-Length') or \
+            headers.get('Content-Range').split('/')[-1]
+        fileModel = ModelImporter.model('file')
+        fileDoc = fileModel.createLinkFile(
+            url=url, parent=parent, name=dataMap.getName(), parentType=parentType,
+            creator=user, size=int(size),
+            mimeType=headers.get('Content-Type', 'application/octet-stream'),
+            reuseExisting=True)
+        gc_file = fileModel.filter(fileDoc, user)
+
+        gc_item = ModelImporter.model('item').load(
+            gc_file['itemId'], force=True)
+        gc_item['meta'] = {'identifier': 'unknown', 'provider': 'HTTP'}
```

Comment:
Since null_provider allows to specify identifier as `None`, we could use that here too for consistency:
```suggestion
        gc_item['meta'] = {'identifier': None, 'provider': 'HTTP'}
```
BTW, this requires changes to `datasetModel` (`server/rest/dataset.py`) and `dataMapSchema` (`server/schema/misc.py`) schemas, where allowed types for fields `identifier` and `doi` respectively, should be changed to `'type': ['string', 'null'],`